src/ReinforcementLearning/SimpleAgent/RandomAgent.h

```diff
@@ -14,26 +15,32 @@ class RandomAgent : public BaseAgent {
     public:
     RandomAgent(){};
     ~RandomAgent(){};
+    std::vector <std::pair<std::vector <bool>, unsigned int>> history;
+    
     /** The member policy selects selectable edges at random.
     */
-    virtual void policy(state* s) const override {
+    virtual int act(state* s) const override {
         std::mt19937 gen(std::chrono::high_resolution_clock::now().time_since_epoch().count());
-        bool hasEdge = true;
-        while (hasEdge) {
-            hasEdge = false;
-            std::vector <int> selectableIndices;
-            for (int j = 0; j < s->selectable.size(); j++) {
-                if (s->selectable[j] == true) {
-                    selectableIndices.push_back(j);
-                    hasEdge = true;
-                }
-            }
-            if (hasEdge == true) {
-                std::uniform_int_distribution<> dis(0, selectableIndices.size() - 1);
-                int d = selectableIndices[dis(gen)];
-                s->select(d);
+        std::vector <int> selectableIndices;
+        bool hasEdge = false;
+        for (int j = 0; j < s->selectable.size(); j++) {
+            if (s->selectable[j] == true) {
+                selectableIndices.push_back(j);
+                hasEdge = true;
             }
         }
+        if (hasEdge == true) {
+            std::uniform_int_distribution<> dis(0, selectableIndices.size() - 1);
+            unsigned int d = selectableIndices[dis(gen)];
+            vector <bool> b1 = s->selectedSubset;
+            std::pair <std::vector <bool>, unsigned int> p1= std::make_pair(b1, d);
+            this->history.push_back(p1);
+            s->select(d); //for tests ONLY!!
+            return d;
+        }
```

Comment:
coding style

Reply:
Not sure what exactly is the mistake here, I think i fixed it?